CONTRIBUTING.md

```diff
@@ -2,6 +2,19 @@
 
 We welcome all contributions!
 
+
+## Requirements
+
+- Ruby version 3.1.1+
```

Comment:
This is only enforced in development, not at runtime. We have a CI task to test on 2.5.0, but the tools we use (e.g. RBS, Steep, Rubocop) all have a dependency on 3+.